File: app/src/main/java/com/example/mvvm_artichecture_sample/data/remote/ApiInterface.java
package com.example.mvvm_artichecture_sample.data.remote;

import com.example.mvvm_artichecture_sample.data.remote.apimodel.ReposnseModel;

import io.reactivex.Observable;
import retrofit2.Response;
import retrofit2.http.GET;
import retrofit2.http.Headers;

public interface ApiInterface {

    @GET("countries")
    Observable<Response<ReposnseModel>> getList();
}

File: app/src/main/java/com/example/mvvm_artichecture_sample/data/remote/ApiInterface.kt
package com.example.mvvm_artichecture_sample.data.remote

import com.example.mvvm_artichecture_sample.data.remote.apimodel.ResponseModel
import kotlinx.coroutines.Deferred
import retrofit2.Response
import retrofit2.http.GET

interface ApiInterface {

    @GET("countries")
    suspend fun getCountryList(): Response<ResponseModel>

    @GET("countries")
    fun getCountryAsync(): Deferred<Response<ResponseModel>>

}
